docs/src/developer/future.md

```diff
@@ -0,0 +1,126 @@
+```@meta
+CurrentModule = Nemo
+```
+
+# Future plans
+
+## Ring and CommRing
+
+Currently all commutative ring types belong to `Ring` and their elements to
+`RingElem` (and `RingElement`) and we have separate types for noncommutative
+rings and elements thereof, i.e. `NCRing` and `NCRingElem` etc.
+
+However, it would be more logical to use `Ring` for not necessarily
+commutative rings and `CommRing`, `CRing` or `CommutativeRing` (the name has
+not been decided on yet) for commutative rings.
+
+This is a big change and should happen with plenty of warning for the
+community. It would be convenient if a script could be made available to
+automate this.
+
+## Matrices and polynomials over number fields
+
+The Antic C library is used for number field elements, but matrices and
+polynomials over these are handled by Julia. There is nothing explicitly
+wrong with this, except that the jit compilation is costly for such basic
+operations and the code is not as easy for others to use in other projects
+if it is not in C.
```

Comment:
> the code is not as easy for others to use in other projects if it is not in C.

It's not obvious why this is so.

Reply:
There are numerous C projects using Antic. They'd have to add a dependency on Julia to use the Julia code. I'm not sure that needs spelling out.

Reply:
I'll add a couple of extra words of clarification.

Reply:
Oh I see, sorry, I was closemindedly thinking of Julia projects... thank you for having added clarifications.